airflow/operators/python.py

```diff
@@ -85,6 +86,9 @@ def __init__(
         templates_exts: Optional[List[str]] = None,
         **kwargs
     ) -> None:
+        if kwargs.get("provide_context"):
+            warnings.warn("provide_context is deprecated as of 2.0 and is no longer required")
+            kwargs.pop('provide_context', None)
```

Comment:
```suggestion
        if kwargs.get("provide_context"):
            warnings.warn("provide_context is deprecated as of 2.0 and is no longer required", DeprecationWarning, stacklevel=2)
            kwargs.pop('provide_context', None)
```

This should take care of @ashb 's suggestion